app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt

```diff
@@ -87,6 +88,24 @@ class KiwixMainActivity : CoreMainActivity() {
     bottom_nav_view.setupWithNavController(navController)
   }
 
+  override fun onConfigurationChanged(newConfig: Configuration) {
+    super.onConfigurationChanged(newConfig)
+    bottom_nav_view?.menu?.apply {
+      findItem(R.id.readerFragment)?.title = resources.getString(R.string.reader)
+      findItem(R.id.libraryFragment)?.title = resources.getString(R.string.library)
+      findItem(R.id.downloadsFragment)?.title = resources.getString(R.string.download)
+    }
+    drawer_nav_view?.menu?.apply {
+      findItem(R.id.menu_bookmarks_list)?.title = resources.getString(R.string.bookmarks)
+      findItem(R.id.menu_history)?.title = resources.getString(R.string.history)
+      findItem(R.id.menu_notes)?.title = resources.getString(R.string.pref_notes)
+      findItem(R.id.menu_host_books)?.title = resources.getString(R.string.menu_wifi_hotspot)
+      findItem(R.id.menu_settings)?.title = resources.getString(R.string.menu_settings)
+      findItem(R.id.menu_help)?.title = resources.getString(R.string.menu_help)
```

Comment:
Instead of this, I'm sure android provided APIs to do this automatically. IMO manually updating each view isn't ideal in this case.

Reply:
hi @gouri-panda ,

Another way is to recreate the activity since bottom navigation bar is attached to activity not fragment and if i am going to recreate it then we will lose the state, I am not getting any better solution , let me know if you have any ?

Reply:
@MohitMaliDeveloper Which state are you talking about? We are storing states in the ViewModel. How this is gonna lose?

Reply:
hi @gouri-panda ,
i'm taking about activity ```navController``` state, if we recreate the activity then it will start from Splash Screen.